### PR TITLE
replace ryslog with rsyslog-forwarder and refactor

### LINKED ISSUE #620: Subordinate charms example references a non-subordinate charm
In the discussion of subordinate charms, rsyslog is listed as an example charm. At the time of writing that may have been true but now rsyslog is a regular charm. rsyslog-forwarder is a subordinate.

A new charm author would reasonably go look at rsyslog to see how subordinates are constructed and be very confused.

### PR DESCRIPTION
**Pull request by pmatulis**
Fixes #620 

:warning: The command `juju add-relation rsyslog-forwarder mysql` will fail:

`ERROR cannot add relation "rsyslog-forwarder:juju-info mysql:juju-info": principal and subordinate applications' series must match`

buy MySQL is not actually required here. The example deployment just looks more authentic with it.

### CODE REVIEW

File: src/en/authors-subordinate-applications.md [hunk 1]
```diff
@@ -147,20 +104,53 @@ juju deploy wordpress
...
+
+## Status of subordinates
+
+The status output contains details about subordinate units under the status of
+the principal application unit that it is sharing the container with. The
+subordinate unit's output matches the formatting of existing unit entries but
+omits machine, public-address and subordinates (which are all the same as the
+principal unit).
+
+Below is shown two excerpts from the output to `juju status --format yaml`
```
Comment (evilnick) [inline src/en/authors-subordinate-applications.md]: Shown below are two...

File: src/en/authors-subordinate-applications.md [hunk 2]
```diff
@@ -147,20 +104,53 @@ juju deploy wordpress
...
-relation to both wordpress and mysql but these new relationships are implemented
-only between the principal unit and the subordinate unit. A subordinate unit
-may still have standard relations established with any unit in its environment
-as usual.
+This will create a rsyslog-forwarder application unit inside each of the
+containers holding the mysql and wordpress units. The rsyslog-forwarder
+application has a standard client-server relation to both wordpress and mysql
+but these new relationships are implemented only between the principal unit and
+the subordinate unit. A subordinate unit may still have standard relations
+established with any unit in its environment as usual.
```
Comment (evilnick) [inline src/en/authors-subordinate-applications.md]: I would avoid the use of 'environment'.

File: src/en/authors-subordinate-applications.md (hunk 1 above)
Comment (pmatulis) [reply src/en/authors-subordinate-applications.md]: OK

File: src/en/authors-subordinate-applications.md (hunk 2 above)
Comment (pmatulis) [reply src/en/authors-subordinate-applications.md]: OK

**Pull request merged**